### PR TITLE
Add UpwindBiasedGradient operator

### PR DESCRIPTION
**Pull request by dennisYatunin**
This PR adds an `UpwindBiasedGradient` operator that can be used in CliMA/ClimaAtmos.jl#3843 (see [Tapio's comment](https://github.com/CliMA/ClimaAtmos.jl/pull/3843#discussion_r2134392568)). This operator computes a first-order gradient with either left biasing or right biasing, depending on the direction of an associated velocity field. A convergence and correctness test has also been added.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: test/Operators/finitedifference/convergence_column.jl [hunk 1]
```diff
@@ -409,6 +409,49 @@ end
...
+    for (k, n) in enumerate(n_elems_seq)
+        domain = Domains.IntervalDomain(
+            Geometry.ZPoint{FT}(-pi),
+            Geometry.ZPoint{FT}(pi);
+            periodic = true,
+        )
+        mesh = Meshes.IntervalMesh(domain; nelems = n)
+
+        center_space = Spaces.CenterFiniteDifferenceSpace(device, mesh)
+        face_space = Spaces.FaceFiniteDifferenceSpace(center_space)
```
Comment (charleskawczynski) [inline test/Operators/finitedifference/convergence_column.jl]:
```suggestion
        face_space = Spaces.face_space(center_space)
```

Comment (charleskawczynski) [reply test/Operators/finitedifference/convergence_column.jl]: I think that this is just a cleaner pattern.

File: test/Operators/finitedifference/convergence_column.jl [hunk 2]
```diff
@@ -409,6 +409,49 @@ end
...
+
+        ᶜw = Geometry.WVector.(map(_ -> rand(), ones(center_space)))
+        ᶠw = Geometry.WVector.(map(_ -> rand(), ones(face_space)))
```
Comment (charleskawczynski) [inline test/Operators/finitedifference/convergence_column.jl]:
```suggestion
        ᶜw = Geometry.WVector.(map(_ -> 2*rand() - 1, ones(center_space)))
        ᶠw = Geometry.WVector.(map(_ -> 2*rand() - 1, ones(face_space)))
```
Shouldn't this also test negative `w`s?

File: test/Operators/finitedifference/convergence_column.jl (hunk 1 above)
Comment (dennisYatunin) [reply test/Operators/finitedifference/convergence_column.jl]: Yeah, the `face_space` version is better. I've updated every instance of the `FaceFiniteDifferenceSpace(center_space)` pattern in this file.
